### DISCUSSION TITLE
Non-rectangular window boundary

### DISCUSSION THREAD

**Discussion by Zoybean** [rust-windowing/winit#3211] [Q&A]
Is there any way to either selectively pass-through mouse events, or otherwise implement a non-rectangular window for the purposes of mouse events?
My specific use-case is to make a draggable desktop creature, and ideally they would have a non-rectangular "hitbox" for clicks, where clicking on a transparent area on the creature would send the click to the underlying window.
I can make a borderless window with transparent sections, and I can see that #1434 added a way to pass-through mouse events with `set_cursor_hittest` (also #2559), but I suspect that this is too coarse-grained to implement something like I want, as I would need to use mouse events to determine whether to enable mouse events.

**Comment by Zoybean** [comment]
As an example of what I'm trying to imitate: [Baba Friend by Hempuli](https://hempuli.itch.io/baba-friend)

**Reply by kchibisov** [reply]
All underlying APIs use just a rect and don't allow forwarding, not on a winit level, but on a windowing system levels. I'm not sure that what you want can function without tight integrations with particular OS of choice.

Maybe there's a windows specific API for that sort of things, as in there was an idea to add `hittest` events for windows, but nothing cross platform.
